model_analyzer/analyzer.py

```diff
@@ -266,25 +301,42 @@ def _profile(self, perf_analyzer, dcgm_monitor):
 
         # Insert all records into aggregator and get aggregated DCGM records
         record_aggregator = RecordAggregator()
-        for record in perf_records + dcgm_records:
+        for record in dcgm_records:
             record_aggregator.insert(record)
-        return record_aggregator.aggregate()
 
-    def _create_output_table(self, aggregation_tag='Max'):
+        records_groupby_gpu = {}
+        for tag in self.dcgm_tags:
```

Comment:
If we move this loop into `RecordAggregator.groupby` we can turn this into
```
records_groupby_gpu = RecordAggregator.groupby(
                             record_aggregator, 
                             self.dcgm_tags, 
                             (lambda record: record.device().device_id()))
```

Reply:
I fixed this in the new commit.